attendance/API_HR/c-hr.php

```diff
@@ -975,6 +975,43 @@ public static function getUserMonthAttendace($userid, $year, $month) {
         return $finalReturn;
     }
 
+    public static function getAllYearAttendance() {
+
+        $r_error = 0;
+        $r_message = "";
+        $return = array();
+        $years = array();
+        $attendance_rows = array();
+
+        $q = "SELECT * from attendance";
+        $runQuery = self::DBrunQuery($q);
+        $rows = self::DBfetchRows($runQuery);
+        
+        for ( $i = 0; $i < count($rows); $i++ ){            
+            $full_date = explode(" ", $rows[$i]['timing']);
+            $explode_full_date = explode("-", $full_date[0]);
+            $year = $explode_full_date[2];            
+            if($year && !in_array($year, $years)){
+                $years[] = $year;
+            }
+        }
+        
+        for( $k = 0; $k < count($years); $k++ ){
```

Comment:
why are you using 2 loops and query, when you already have all rows of this table
you can extract the year from itself